athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/VertexRowWriter.java

```diff
@@ -157,4 +182,24 @@ public static void writeRowTemplate(RowWriterBuilder rowWriterBuilder, Field fie
                 break;
         }
     }
+
+    private static Map<String, Object> contextAsMap(Object context, boolean caseInsensitive) 
+    {
+        Map<String, Object> contextAsMap = (Map<String, Object>) context;
+
+        if (!caseInsensitive) {
+            return contextAsMap;
+        }
+
+        Object fieldValueID = contextAsMap.get(T.id);
+        if (fieldValueID != null) {
+            contextAsMap.remove(T.id);
+            contextAsMap.remove(T.label);
+            contextAsMap.put(SpecialKeys.ID.toString(), fieldValueID);
```

Comment:
Same question here as above in regards to mutating `Object context`

Reply:
same as above, my initial thought was it would be straight forward but it fails because the keys are not Strings